File: src/2025-02-23-deepclaude-try/go/exercises/practice/wordy/wordy.go
package wordy

import (
	"strconv"
	"strings"
)

func Answer(question string) (int, bool) {
	if !strings.HasPrefix(question, "What is ") || !strings.HasSuffix(question, "?") {
		return 0, false
	}

	expr := strings.TrimSpace(strings.TrimSuffix(strings.TrimPrefix(question, "What is "), "?"))
	if expr == "" {
		return 0, false
	}

	words := strings.Fields(expr)
	var tokens []string

	for i := 0; i < len(words); {
		// Check for two-word operators first
		if i+1 < len(words) {
			if (words[i] == "multiplied" && words[i+1] == "by") || (words[i] == "divided" && words[i+1] == "by") {
				tokens = append(tokens, words[i]+" "+words[i+1])
				i += 2
				continue
			}
		}

		// Check single-word operators
		if words[i] == "plus" || words[i] == "minus" {
			tokens = append(tokens, words[i])
			i++
			continue
		}

		// Check if it's a number
		if _, err := strconv.Atoi(words[i]); err == nil {
			tokens = append(tokens, words[i])
			i++
			continue
		}

		// Invalid token
		return 0, false
	}

	if len(tokens) == 0 {
		return 0, false
	}

	// Validate and parse initial number
	initialNum, err := strconv.Atoi(tokens[0])
	if err != nil {
		return 0, false
	}

	if len(tokens) == 1 {
		return initialNum, true
	}

	// Validate operator-number pairs
	for i := 1; i < len(tokens); i += 2 {
		// Check operator validity
		op := tokens[i]
		if op != "plus" && op != "minus" && op != "multiplied by" && op != "divided by" {
			return 0, false
		}

		// Check number existence and validity
		if i+1 >= len(tokens) {
			return 0, false
		}
		numStr := tokens[i+1]
		if _, err := strconv.Atoi(numStr); err != nil {
			return 0, false
		}
	}

	// Perform calculations
	result := initialNum
	for i := 1; i < len(tokens); i += 2 {
		op := tokens[i]
		num, _ := strconv.Atoi(tokens[i+1]) // already validated above

		switch op {
		case "plus":
			result += num
		case "minus":
			result -= num
		case "multiplied by":
			result *= num
		case "divided by":
			result /= num
		}
	}

	return result, true
}

File: src/2025-02-23-deepclaude-try/javascript/exercises/practice/wordy/wordy.js
export const answer = (question) => {
  // Validate question format
  if (!question || typeof question !== 'string') {
    throw new Error('Invalid question format');
  }

  const match = question.match(/^What is (.*)\?$/i);
  if (!match) {
    throw new Error('Unknown operation');
  }

  // Extract the expression
  const expression = match[1].trim();
  if (!expression) {
    throw new Error('Syntax error');
  }

  // Split into tokens and filter empty strings
  const tokens = expression
    .split(/(plus|minus|multiplied by|divided by)/i)
    .map(token => token.trim())
    .filter(Boolean);

  // Validate first token is a number
  if (!/^-?\d+$/.test(tokens[0])) {
    throw new Error('Syntax error');
  }

  let result = parseInt(tokens[0]);

  // Process operations from left to right
  for (let i = 1; i < tokens.length; i += 2) {
    const operator = tokens[i].toLowerCase();
    const nextNum = tokens[i + 1];

    // Validate next token exists and is a number
    if (!nextNum || !/^-?\d+$/.test(nextNum)) {
      throw new Error('Syntax error');
    }

    const num = parseInt(nextNum);

    switch (operator) {
      case 'plus':
        result += num;
        break;
      case 'minus':
        result -= num;
        break;
      case 'multiplied by':
        result *= num;
        break;
      case 'divided by':
        result /= num;
        break;
      default:
        throw new Error('Unknown operation');
    }
  }

  return result;
};
